src/chainLinks/bugsnag-link.js

```diff
@@ -0,0 +1,90 @@
+const ChainLink = require('./chain-link');
+const bugsnag = require('bugsnag');
+
+/**
+  @class BugsnagLink
+  A Bugsnag notification chain link
+  @implements ChainLink @class - a basic interface for each link based on the Chain of Resp Pattern
+  This chain link is responsible for firing a notification to the bugsnag endpoint
+
+  Has the following configurations (either env var or settings param):
+  - BUGSNAG_LOGGING {'true'|'false'} - switches on / off the use of this chain link
+  - MIN_LOG_LEVEL_BUGSNAG = {'silly'|'verbose'|'debug'|'info'|'warn'|'error'} - min log level of a message to notify
+  This config has a higher priority than a global DEFAULT_LOG_LEVEl config
+  @see ChainLink @class for info on the log level priorities
+  If a message's level is >= than a MIN_LOG_LEVEL_BUGSANG - it will be notified. Otherwise - skipped
+
+  Environment variables have a higher priority over a settings object parameters
+**/
+class BugsnagLink extends ChainLink {
+  /**
+    @constructor
+    Construct an instance of a BugsnagLink @class
+    @param settings {Object} - LoggerChain configuration object
+    @param nextChainLink {Object} - Optional parameter. A next object on the chain
+  **/
+  constructor(settings, nextChainLink) {
+    super(nextChainLink, settings);
+    if (this.settings.BUGS_TOKEN) {
+      bugsnag.register(this.settings.BUGS_TOKEN, {
+        releaseStage: process.env.NODE_ENV || 'local',
+        notifyReleaseStages: ['production', 'staging']
```

Comment:
@mrister yes, but I think that since it is more related to this particular chainLink, I would add it when decoupling it, so I did not add it. But of course I can do that